lnst/Recipes/ENRT/MeasurementGenerators/LinuxPerfMeasurementGenerator.py

```diff
@@ -18,24 +18,26 @@ class LinuxPerfMeasurementGenerator(BaseMeasurementGenerator):
     def generate_perf_measurements_combinations(self, config):
         combinations = super().generate_perf_measurements_combinations(config)
 
-        if not self.params.do_linuxperf_measurement:
-            return combinations
-
-        # create linuxperf data folder
-        linuxperf_data_folder: str = os.path.abspath(
-            os.path.join(self.current_run.log_dir, "linuxperf-data")
-        )
-        try:
-            os.mkdir(linuxperf_data_folder)
-        except FileExistsError:
-            pass
+        if self.params.do_linuxperf_measurement:
+            # create linuxperf data folder
+            linuxperf_data_folder: str = os.path.abspath(
+                os.path.join(self.current_run.log_dir, "linuxperf-data")
+            )
+            try:
+                os.mkdir(linuxperf_data_folder)
+            except FileExistsError:
+                pass
 
         for combination in combinations:
-            measurement: BaseMeasurement = LinuxPerfMeasurement(
-                self.matched,
-                self.linuxperf_cpus,
-                data_folder=linuxperf_data_folder,
-                recipe_conf=config,
-            )
+            if self.params.do_linuxperf_measurement:
```

Comment:
I'd personally add a type hint for `res` before this line, if we know that only `BaseMeasurement` objects will be present in the list.
```py
res: list[BaseMeasurement]
```
Everything else LGTM, learning something new every day.

Reply:
added, for consistency with the rest of the method.

Reply:
@olichtne The type hint should be placed before the code flow branches into the `if-else`. Currently the second type definition throws `Name "res" already defined.`

```py
res: list[BaseMeasurement]
if self.params.do_linuxperf_measurement:
    ...
    res = [measurement] + combination
else:
    res = combination```

Reply:
ok, i don't get these errors so I didn't notice...